### PR TITLE
Add `AWS S3 Bucket Versioning Disable` Rule

### PR DESCRIPTION
**Pull request by sj-sec**
### Summary of the Pull Request

This detection alerts when a threat actor disable AWS S3 bucket versioning. This tactic prohibits victims from restoring any impacted S3 objects and is a common technique used in real world AWS S3 ransomware and extortion incidents. 

### Changelog

new: AWS S3 Bucket Versioning Disable

### Example Log Event
```json
{
    "eventVersion": "1.09",
    "userIdentity": {
        "type": "IAMUser",
        "principalId": "AIDA[REDACTED]",
        "arn": "arn:aws:iam::[REDACTED]:user/Admin",
        "accountId": "[REDACTED]",
        "accessKeyId": "[REDACTED]",
        "userName": "Admin",
        "sessionContext": {
            "attributes": {
                "creationDate": "2023-10-27T13:59:36Z",
                "mfaAuthenticated": "true"
            }
        }
    },
    "eventTime": "2023-10-27T14:01:46Z",
    "eventSource": "[s3.amazonaws.com](https://urldefense.proofpoint.com/v2/url?u=http-3A__s3.amazonaws.com&d=DwMFaQ&c=V9IgWpI5PvzTw83UyHGVSoW3Uc1MFWe5J8PTfkrzVSo&r=S2f_KC8VUGBUG_uY6EuH0nwTqTPUO63PLSzVmyfn9Dg&m=qyrgyDrpytajkt2gsxejVmNho2E8PoaagvyYKOylN_nivZHXPZs7VDSSoxC7ZUjP&s=uOvIyii76sg7ajPZHnoo4IZViznZguiVvrysMjH9gJ8&e=)",
    "eventName": "PutBucketVersioning",
    "awsRegion": "us-east-1",
    "sourceIPAddress": "[REDACTED]",
    "userAgent": "[S3Console/0.4, aws-internal/3 aws-sdk-java/1.12.488 Linux/5.10.196-163.743.amzn2int.x86_64 OpenJDK_64-Bit_Server_VM/25.372-b08 java/1.8.0_372 vendor/Oracle_Corporation cfg/retry-mode/standard]",
    "requestParameters": {
        "bucketName": "[REDACTED]",
        "Host": "[s3.amazonaws.com](https://urldefense.proofpoint.com/v2/url?u=http-3A__s3.amazonaws.com&d=DwMFaQ&c=V9IgWpI5PvzTw83UyHGVSoW3Uc1MFWe5J8PTfkrzVSo&r=S2f_KC8VUGBUG_uY6EuH0nwTqTPUO63PLSzVmyfn9Dg&m=qyrgyDrpytajkt2gsxejVmNho2E8PoaagvyYKOylN_nivZHXPZs7VDSSoxC7ZUjP&s=uOvIyii76sg7ajPZHnoo4IZViznZguiVvrysMjH9gJ8&e=)",
        "versioning": "",
        "VersioningConfiguration": {
            "Status": "Suspended",
            "xmlns": "[http://s3.amazonaws.com/doc/2006-03-01/](https://urldefense.proofpoint.com/v2/url?u=http-3A__s3.amazonaws.com_doc_2006-2D03-2D01_&d=DwMFaQ&c=V9IgWpI5PvzTw83UyHGVSoW3Uc1MFWe5J8PTfkrzVSo&r=S2f_KC8VUGBUG_uY6EuH0nwTqTPUO63PLSzVmyfn9Dg&m=qyrgyDrpytajkt2gsxejVmNho2E8PoaagvyYKOylN_nivZHXPZs7VDSSoxC7ZUjP&s=CHR1bWGV-P1QAtDRG7pEhI2DNm23v1zdaPCl10LEerU&e=)"
        }
    },
    "responseElements": null,
    "additionalEventData": {
        "SignatureVersion": "SigV4",
        "CipherSuite": "ECDHE-RSA-AES128-GCM-SHA256",
        "bytesTransferredIn": 125,
        "AuthenticationMethod": "AuthHeader",
        "x-amz-id-2": "liRp5W6YkgGvinnVqvyON6Q/pAEnWT3/f9p25Upq5etFUWX34fwDsMONKO/Ou8FQMti1bPSzmOY=",
        "bytesTransferredOut": 0
    },
    "requestID": "10S4QJVQS0R6ZNDP",
    "eventID": "5876d44b-1d20-4c35-9e9b-e33cfd1df573",
    "readOnly": false,
    "resources": [
        {
            "accountId": "[REDACTED]",
            "type": "AWS::S3::Bucket",
            "ARN": "[REDACTED]"
        }
    ],
    "eventType": "AwsApiCall",
    "managementEvent": true,
    "recipientAccountId": "[REDACTED]",
    "vpcEndpointId": "vpce-f40dc59d",
    "eventCategory": "Management",
    "tlsDetails": {
        "tlsVersion": "TLSv1.2",
        "cipherSuite": "ECDHE-RSA-AES128-GCM-SHA256",
        "clientProvidedHostHeader": "[s3.amazonaws.com](https://urldefense.proofpoint.com/v2/url?u=http-3A__s3.amazonaws.com&d=DwMFaQ&c=V9IgWpI5PvzTw83UyHGVSoW3Uc1MFWe5J8PTfkrzVSo&r=S2f_KC8VUGBUG_uY6EuH0nwTqTPUO63PLSzVmyfn9Dg&m=qyrgyDrpytajkt2gsxejVmNho2E8PoaagvyYKOylN_nivZHXPZs7VDSSoxC7ZUjP&s=uOvIyii76sg7ajPZHnoo4IZViznZguiVvrysMjH9gJ8&e=)"
    }
}
```

### Fixed Issues
N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Comment by nasbench** [comment]
Hey @sj-sec and thanks for this first contribution and welcome to the Sigma community :)

I'll review the rule in a bit. Also don't worry about the test failing, I'll get that fixed too.

Cheers.

**Comment by sj-sec** [comment]
Thanks @nasbench happy to be here! I have plenty more AWS rules to submit :)

**Review by nasbench — APPROVED** [review]

File: rules/cloud/aws/cloudtrail/aws_disable_bucket_versioning.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+date: 2023/10/28
+tags:
+    - attack.impact
+    - attack.t1490
+logsource:
+    product: aws
+    service: cloudtrail
+detection:
+    selection:
+        eventSource: s3.amazonaws.com
```
Comment (frack113) [inline rules/cloud/aws/cloudtrail/aws_disable_bucket_versioning.yml:17]:
```suggestion
        eventSource|contains: s3.amazonaws.com
```
From the log 
```json
"eventSource": "[s3.amazonaws.com](https://urldefense.proofpoint.com/v2/url?u=http-3A__s3.amazonaws.com&d=DwMFaQ&c=V9IgWpI5PvzTw83UyHGVSoW3Uc1MFWe5J8PTfkrzVSo&r=S2f_KC8VUGBUG_uY6EuH0nwTqTPUO63PLSzVmyfn9Dg&m=qyrgyDrpytajkt2gsxejVmNho2E8PoaagvyYKOylN_nivZHXPZs7VDSSoxC7ZUjP&s=uOvIyii76sg7ajPZHnoo4IZViznZguiVvrysMjH9gJ8&e=)",
```

**Review by frack113 — CHANGES_REQUESTED** [review]

**Review by frack113 — APPROVED** [review]

**Review by frack113 — APPROVED** [review]

**Comment by nasbench** [comment]
Thanks once again @sj-sec for this contribution and can't wait to see more AWS goodness from you :)

**Pull request merged**